habitat-lab/habitat/tasks/rearrange/actions/oracle_arm_action.py

```diff
@@ -155,30 +153,31 @@ def set_desired_ee_pos(self, ee_pos: np.ndarray) -> None:
             self.cur_articulated_agent.fix_joint_values = des_joint_pos
 
     def step(self, pick_action, **kwargs):
-        
         object_pick_pddl_idx = pick_action[0]
         should_pick = pick_action[1]
 
-        if object_pick_pddl_idx <= 0 or object_pick_pddl_idx > len(self._entities):
+        if object_pick_pddl_idx > len(self._entities):
             return self.ee_target
 
-        object_coord = self._get_coord_for_pddl_idx(object_pick_pddl_idx)
-        cur_ee_pos = self.cur_articulated_agent.ee_transform().translation
-        # TODO(zxz): modify translation matrix
-        translation = object_coord - cur_ee_pos
-
-        # translation from object to end effector in base frame
-        translation = self.cur_articulated_agent.base_transformation.inverted().transform_vector(translation)
-
-
-        should_rest = False
-        # if self.hand_state == HandState.APPROACHING:  # Approaching
-        # Only move the hand to object if has to drop or object is not grabbed
-        if should_pick == 0 or self.cur_grasp_mgr.snap_idx is None:
-            translation = np.clip(translation, -1, 1)
+        if should_pick == 1:
+            # or self.cur_grasp_mgr.snap_idx is None
+            object_coord = self._get_coord_for_pddl_idx(object_pick_pddl_idx)
+            cur_ee_pos = self.cur_articulated_agent.ee_transform().translation
+            # TODO(zxz): 使用ee_transform获得的ee_pos与arm_joint_pos使用calc_fk计算的ee_pos不一样
```

Comment:
Has this problem been solved with the latest 3D-2D projection fix in main branch?

Reply:
I think this is OK for navigation sensor. However, there are some issues with the arm_workspace_sensor. I am trying to figure out the source of the bugs. Whether it is from the is_reachable or the projection.